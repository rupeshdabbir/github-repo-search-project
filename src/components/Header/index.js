// @flow

import React from 'react';
import {
  StyleSheet,
  css,
} from 'aphrodite/no-important';
import SearchForm from 'components/SearchForm';
import Logo from 'components/Logo';
import {viewport} from 'theme';
import 'suitcss-utils-flex/lib/flex-sm.css';

type Props = {
  onSubmit: Function,
  searchTerm: string,
};

export default function Header({onSubmit, searchTerm}: Props) {
  return (
    <div
      className={`${css(styles.Header)} u-sm-flex u-sm-flexJustifyBetween u-sm-flexAlignItemsCenter`}
    >
      <div className="u-sm-flexGrow2">
        <Logo />
      </div>
      <div className={`${css(styles.Header_form)} u-sm-flexGrow1`}>
        <SearchForm
          onSubmit={onSubmit}
          initialInputValue={searchTerm}
        />
      </div>
    </div>
  );
}

const styles = StyleSheet.create({
  Header: {
    paddingTop: 15,
    paddingBottom: 15,
  },

  Header_form: {
    marginTop: 20,

    [viewport.SM]: {
      margin: 0,
    },
  },
});
