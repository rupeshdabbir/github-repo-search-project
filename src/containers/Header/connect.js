// @flow

import { connect } from 'react-redux';

export function mapDispatchToProps(dispatch: Function) {
  return {
    searchForRepos(search: string) {
      dispatch({
        type: 'SEARCH_REQUEST',
        payload: {
          search,
        },
      });
    },
  };
}

export default connect(null, mapDispatchToProps);
