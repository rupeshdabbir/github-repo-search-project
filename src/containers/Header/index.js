// @flow

import React, { Component } from 'react';
import { StyleSheet } from 'aphrodite/no-important';
import Header from 'components/Header';
import Container from 'components/Container';
import connect from './connect';

type Props = {
  onSubmit: Function,
  searchQuery: string,
  searchTerm: string,
  searchForRepos: Function,
};

export class HeaderContainer extends Component {
  static defaultProps = {
    searchTerm: '',
  };

  constructor(props: Props) {
    super(props);
    this.handleSearchRepos(props.searchQuery);
  }

  componentWillReceiveProps(nextProps: Props) {
    if (nextProps.searchQuery !== this.props.searchQuery) {
      this.handleSearchRepos(nextProps.searchQuery);
    }
  }

  handleSearchRepos(searchQuery: string): void {
    if (!searchQuery) {
      return;
    }
    this.props.searchForRepos(searchQuery);
  }

  render() {
    const { onSubmit, searchTerm } = this.props;

    return (
      <Container rootStyle={styles.HeaderContainer}>
        <Header onSubmit={onSubmit} searchTerm={searchTerm} />
      </Container>
    );
  }
}

const styles = StyleSheet.create({
  HeaderContainer: {
    backgroundColor: '#f4f4f4',
    borderBottom: '1px solid #aaa',
    boxShadow: '0px 0px 2px 1px rgba(0, 0, 0, .2)',
  },
});

export default connect(HeaderContainer);
