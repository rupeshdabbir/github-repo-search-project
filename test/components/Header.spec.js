import React from 'react';
import renderer from 'react-test-renderer';
import {
  StyleSheetTestUtils,
} from 'aphrodite';
import {
  StaticRouter as Router,
} from 'react-router-dom';

import Header from 'components/Header';

describe('Component: Header', () => {

  beforeAll(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterAll(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('should render', () => {
    const component = renderer.create(
      <Router context={{}}>
        <Header onSubmit={jest.fn()} searchTerm="test">
          <div>inner component</div>
        </Header>
      </Router>
    );
    expect(component).toMatchSnapshot();
  });

});
