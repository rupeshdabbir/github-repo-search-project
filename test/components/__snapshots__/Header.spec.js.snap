// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Header should render 1`] = `
<div
  className="Header_1mnqmvf u-sm-flex u-sm-flexJustifyBetween u-sm-flexAlignItemsCenter"
>
  <div
    className="u-sm-flexGrow2"
  >
    <div>
      <h1
        className="Logo_title_229208"
      >
        <a
          className="Logo_link_18deiwy u-flex"
          href="/"
          onClick={[Function]}
        >
          <svg />
          GitHub Repo Search
        </a>
      </h1>
      <p
        className="Logo_subTitle_1r70osc"
      >
        Browse repos via 
        <a
          href="https://developer.github.com/v3/"
        >
          the GitHub API
        </a>
      </p>
    </div>
  </div>
  <div
    className="Header_form_1m5hoq2 u-sm-flexGrow1"
  >
    <form
      className="SearchForm_e296pg"
      onSubmit={[Function]}
    >
      <label
        className="u-hiddenVisually"
        htmlFor="searchInput"
      >
        Search for a repo
      </label>
      <input
        className="SearchForm_input_12zt507"
        id="searchInput"
        onChange={[Function]}
        placeholder="Search for a repo"
        type="text"
        value="test"
      />
      <div
        className="Dropdown-root"
      >
        <div
          aria-haspopup="listbox"
          className="Dropdown-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="Dropdown-placeholder is-selected"
          >
            JavaScript
          </div>
          <div
            className="Dropdown-arrow-wrapper"
          >
            <span
              className="Dropdown-arrow"
            />
          </div>
        </div>
      </div>
      <button
        className="SearchForm_btn_13vgzd5"
        type="submit"
      >
        <svg />
        <span
          className="u-hiddenVisually"
        >
          Submit
        </span>
      </button>
    </form>
  </div>
</div>
`;
